### PR TITLE
Updates to UI testing and Code Coverage Script

### LINKED ISSUE #161: Increased Test Coverage for UI elements
--------
Issue Description
--------
##### What did you expect to see happen?

The test coverage of RAVEN should be above 80%

##### What did you see instead?

The test coverage is currently less than 80% as none of the UI components are currently being tested by the regression system.

##### Do you have a suggested fix for the development team?

The regression test system should at least load and exit the UI elements so that we can test parts of these files. This could be done as a proof of concept first and then followed up with more robust testing of the UI components until full coverage is reached.

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

No error.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed? *Improvement made*
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.) *Increases testing*
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)? *No email needed*
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)? *No stable branch*
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided? *Will be closed with merge*

Close reviewed by @joshua-cogliati-inl

### PR DESCRIPTION
**Pull request by maljovec**
--------
Pull Request Description
--------
##### What issue does this change request address?

Closes #161

##### What are the significant changes in functionality due to this change request?

Fixes a few places that were still using PySide-style API calls to Qt. Also, adds the usage of Xvfb to the `check_py_coverage.sh` script file so that it can test the UI components in a headless environment.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code. *Done*
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts). *No changes*
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details. *Yes*
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass. *Yes*
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.  *This is primarily a code coverage increase change*
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync. *No*
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.  *Done*
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added? *No change*

### CODE REVIEW

File: check_py_coverage.sh [hunk 1]
```diff
@@ -87,10 +87,18 @@ coverage run $EXTRA $FRAMEWORK_DIR/Driver.py test_maap5_code_interface_adaptive_
 coverage run $EXTRA $FRAMEWORK_DIR/Driver.py test_maap5_code_interface_det_multibranch.xml interfaceCheck
 coverage run $EXTRA $FRAMEWORK_DIR/Driver.py test_relap5_code_inss.xml interfaceCheck
 # code interface tests END
+
+Xvfb :8888 &
```
Comment (joshua-cogliati-inl) [inline check_py_coverage.sh]: This should probably be ifed:
```
if which Xvfb; then echo found; else echo not found; fi
```

Comment (joshua-cogliati-inl) [inline check_py_coverage.sh]: only, besides just echoing if it is found or not, do the code to start it and change the DISPLAY

Comment (maljovec) [inline check_py_coverage.sh]: Makes sense to me.

**Comment by brianmoose** [comment]
Sorry, the CIVET recipe didn't sync properly.
It has been fixed and should work in the future.
I did the sync manually this time, the coverage can be viewed at:
http://mooseframework.org/docs/PRs/243/python-coverage/

File: scripts/TestHarness/testers/RavenUtils.py [hunk 2]
```diff
@@ -63,7 +64,8 @@ def __lookUpPreferredVersion(name):
              ("h5py",__lookUpPreferredVersion("h5py")),
              ("scipy",__lookUpPreferredVersion("scipy")),
              ("scikit-learn",__lookUpPreferredVersion("sklearn")),
-             ("matplotlib",__lookUpPreferredVersion("matplotlib"))]
+             ("matplotlib",__lookUpPreferredVersion("matplotlib")),
+             ("pyside",__lookUpPreferredVersion("pyside"))]
```
Comment (joshua-cogliati-inl) [inline scripts/TestHarness/testers/RavenUtils.py]: Either we need to fix bug where pip install command does 'pyside== '  when there is no preferred version, or just remove it from the __pipList

**Comment by alfoa** [comment]
news?

**Comment by maljovec** [comment]
I have tested this on the moosebuild virtual machine and it appears that version 1.5.1 that we currently use of matplotlib available through conda has an incompatibility with the default PySide version. I have switched the recommended version of matplotlib to 1.5.3 which actually downgraded the PySide version. I am unsure at this time which change is more relevant. Hopefully, I can update quickly here.

**Comment by maljovec** [comment]
I believe this PR is ready for review. This does not modify the minimum version of matplotlib (only the preferred version), but given the length of time this has been open, I would prefer to experiment with that separately, since this now appears to be in a working state. In this way, we can in parallel start development of the minimum code coverage attribute for pull requests.

**Pull request merged**